apps/ledger-live-mobile/src/components/QueuedDrawer.tsx

```diff
@@ -0,0 +1,206 @@
+import React, { useCallback, useEffect, useState } from "react";
+import { StyleProp, ViewStyle } from "react-native";
+import { BottomDrawer } from "@ledgerhq/native-ui";
+import { useFocusEffect } from "@react-navigation/native";
+import type { BaseModalProps } from "@ledgerhq/native-ui/components/Layout/Modals/BaseModal";
+import { useSelector } from "react-redux";
+import { isModalLockedSelector } from "../reducers/appstate";
+import { Merge } from "../types/helpers";
+
+export type Props = Merge<
+  BaseModalProps,
+  {
+    isRequestingToBeOpened: boolean;
+    style?: StyleProp<ViewStyle>;
+    containerStyle?: StyleProp<ViewStyle>;
+  }
+>;
+
+/**
+ * A drawer taking into account the currently displayed drawer and other drawers waiting to be displayed.
+ *
+ * This is made possible thanks to a queue of drawers waiting to be displayed. Once the currently displayed drawer
+ * is not displayed anymore (hidden), the next drawer in the queue is notified, and it updates its state
+ * to be make itself visible.
+ *
+ * Setting to true the isRequestingToBeOpened prop will add the drawer to the queue. Setting to false will remove it from the queue.
+ *
+ * The queue is cleaned when its associated screen loses its navigation focus.
+ *
+ * Note: avoid conditionally render this component. Always render it and use isRequestingToBeOpened to control its visibility.
+ * Note: to avoid a UI glitch on Android, do not put this drawer inside NavigationScrollView (and probably any other ScrollView)
+ *
+ * @param isRequestingToBeOpened: to use in place of isOpen. Setting to true will add the drawer to the queue.
+ *   Setting to false will remove it from the queue.
+ * @param onClose: when the user closes the drawer (by clicking on the backdrop or the close button) + when the drawer is hidden
+ *   (this is currently due to a legacy behavior of the BaseModal component. It might change in the future)
+ * @param onModalHide: when the drawer is fully hidden
+ * @param noCloseButton: whether to display the close button or not
+ * @param preventBackdropClick: whether to prevent the user from closing the drawer by clicking somewhere on the screen
+ *   or on the back button
+ * @params all other props are passed to the BaseModal component. Double check the behavior of the props you pass
+ *   before using them in production. Do not use the isOpen prop.
+ */
+const QueuedDrawer = ({
+  isRequestingToBeOpened,
+  onClose,
+  onModalHide,
+  noCloseButton,
+  preventBackdropClick,
+  style,
+  containerStyle,
+  children,
+  ...rest
+}: Props) => {
+  const modalLock = useSelector(isModalLockedSelector);
+  // Actual state that choses if the drawer is displayed or not
+  const [isDisplayed, setIsDisplayed] = useState(false);
+  const [onCloseAlreadyCalled, setOnCloseAlreadyCalled] = useState(false);
+
+  // Makes sure that the drawer system is cleaned when navigating to a new (or back to a) screen
+  useFocusEffect(
+    useCallback(() => {
```

Comment:
Are we sure that we do not have at least one drawer in the app that needs to be kept open while navigating in the background?

Reply:
AFAIK, currently this behavior does not exist: you can't navigate to another screen while a drawer stays open

Reply:
Ah good to know then, thanks!